library/std/src/sys/locks/rwlock/queue.rs

```diff
@@ -200,12 +200,12 @@ impl Node {
     }
 
     /// Prepare this node for waiting.
-    fn prepare(&mut self) {
+    fn prepare(&self) {
         // Fall back to creating an unnamed `Thread` handle to allow locking in
         // TLS destructors.
         self.thread
             .get_or_init(|| thread_info::current_thread().unwrap_or_else(|| Thread::new(None)));
-        self.completed = AtomicBool::new(false);
+        unsafe { self.completed.as_ptr().write(false) }; // can't have a race here
```

Comment:
I should probably either make this a relaxed store, or make the function `unsafe`.